tpm2/encoding_test.go

```diff
@@ -477,3 +477,36 @@ func TestSignEncode(t *testing.T) {
 		}
 	})
 }
+
+func TestTPMLDigestEncode(t *testing.T) {
+	hashA := []byte{0, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 255}
+	hashB := []byte{0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 255}
+
+	d1 := BDigest{
+		Size:   uint16(20), // len 2
+		Digest: hashA,      // len 20
+	}
+	d2 := BDigest{
+		Size:   uint16(20), //len 2
+		Digest: hashB,      //len 20
+	}
+
+	hl := TPMLDigest{
+		Count:   uint32(2),         //len 4
+		Digests: []BDigest{d1, d2}, // 2+20+2+20
+	} //sum 48
+
+	exp := []byte{0, 0, 0, 2, 0, 20, 0, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 255,
+		0, 20, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 255}
+
+	b, err := hl.Encode()
+	if err != nil {
+		t.Errorf("Conversion to []byte failed")
+	}
+	if len(b) != int(48) {
```

Comment:
Should we use len(exp) instead of hardcoded 48 here?

Reply:
agreed